### ISSUE TITLE
Adding an extra Authorization header

### ISSUE THREAD

**Issue by Cinezaster** [sitegui/nodejs-websocket#19]
I'll have to connect to a secure Websocket that uses Basic http authorization.
The only way to do so is to add `Authorization` to the request header. 
```
'Authorization: Basic ' + new Buffer('name'+ ':' + 'password').toString('base64')
```
As for now I changed the header string in the `startHandshake` function.

Would be nice if we could add extra header properties to the ws.connect option?

**Comment by sitegui** [comment]
Uhm, seems nice! I'm open to a pull request if you will ;)

(Otherwise, I can work on it next week)

**Comment by Cinezaster** [comment]
Ok that is what I wanted to hear. :) 

I thing it should be added in the options object of `ws.connect(URL, [options], [callback])`?
This options object is going straight to the `tls.connect(options)` or `net.connect(options)`. 
I suggest to call the argument: `extraHeaders`, this argument is not being used by  `tls` or `net` options object. 
And retrieve these extraHeaders in the `startHandshake` function since the socket is also available within this function.

An extra function in connection.js that creates the header string out of an object could also be a cleaner approach for building the header? I would call it `buildHeaders` and this could be used in `startHandshake` and `answerHandshake `

On the other hand we're making the socket object dirty, with none standard properties? Another cleaner approach might be useful. Can't think of one know.

**Comment by sitegui** [comment]
I like your suggestions of adding it to the `options` argument in `ws.connect(URL, [options], [callback])` and calling it `extraHeaders`.

> And retrieve these extraHeaders in the startHandshake function since the socket is also available within this function.

I don't think the socket stores the raw `options` object given to it (at least I couldn't find it in the docs). Either way, I believe it may be clearer to pass it to `Connection` constructor as an explicit parameter. This would also solve the dirty socket object problem with non-standard properties.

> An extra function in connection.js that creates the header string out of an object could also be a cleaner approach for building the header?
Absolutely!

I'm not sure the `extraHeaders` option would also apply to servers (created by `ws.createServer([options], [callback])`). For simplicity now, we'd better leave it out.

**Comment by sitegui** [comment]
Fixed by #20
